src/components/IOUConfirmationList.js

```diff
@@ -109,34 +149,35 @@ class IOUConfirmationList extends Component {
      */
     getSections() {
         const sections = [];
-
         if (this.props.hasMultipleParticipants) {
             const formattedMyPersonalDetails = getIOUConfirmationOptionsFromMyPersonalDetail(
                 this.props.myPersonalDetails,
-                this.props.numberFormat(this.calculateAmount() / 100, {
+                this.props.numberFormat(this.calculateAmount(this.state.selectedParticipants) / 100, {
                     style: 'currency',
                     currency: this.props.selectedCurrency.currencyCode,
                 }),
             );
 
-            const formattedParticipants = getIOUConfirmationOptionsFromParticipants(this.props.participants,
-                this.props.numberFormat(this.calculateAmount() / 100, {
-                    style: 'currency',
-                    currency: this.props.selectedCurrency.currencyCode,
-                }));
-
             sections.push({
                 title: this.props.translate('iOUConfirmationList.whoPaid'),
                 data: [formattedMyPersonalDetails],
                 shouldShow: true,
                 indexOffset: 0,
             });
+
             sections.push({
                 title: this.props.translate('iOUConfirmationList.whoWasThere'),
-                data: formattedParticipants,
+                data: this.state.selectedParticipants,
                 shouldShow: true,
                 indexOffset: 0,
             });
+
+            sections.push(({
+                title: undefined,
+                data: this.state.unselectedParticipants,
+                shouldShow: !_.isEmpty(this.state.unselectedParticipants),
+                indexOffset: 0,
+            }));
```

Comment:
I think we can clean this up slightly by having a single `push` statement. i.e
```js
sections.push({ }, { }, { })
```

Though whether this change is made depends on your response to the big comment below. :)